modules/activiti-engine/src/main/java/org/activiti/engine/impl/bpmn/behavior/IntermediateCatchEventActivityBehavior.java

```diff
@@ -12,13 +12,31 @@
  */
 package org.activiti.engine.impl.bpmn.behavior;
 
+import java.util.List;
+
+import org.activiti.engine.impl.context.Context;
+import org.activiti.engine.impl.persistence.entity.EventSubscriptionEntity;
+import org.activiti.engine.impl.persistence.entity.ExecutionEntity;
 import org.activiti.engine.impl.pvm.delegate.ActivityExecution;
 
 
 public class IntermediateCatchEventActivityBehavior extends AbstractBpmnActivityBehavior {
 
-
+  @Override
   public void execute(ActivityExecution execution) throws Exception {
+    
+    // Find active subscriptions
+    List<EventSubscriptionEntity> subscriptions = ((ExecutionEntity)execution).getEventSubscriptions();
```

Comment:
The code block added here (lines 29 to 38) is identical to the code added for `EventBasedGatewayActivityBehavior`, excepted by the fact that the first one has a `break` clause. Is this difference intentional? If not, this code could be extracted into a separated class which is injected where necessary to avoid logic duplication. 
In the case where injecting a class is too intrusive in the current code structure you can imagine to have a method in a common super class as well.

What do you think?

Reply:
@erdemedeiros The break statement in `EventBasedGatewayActivityBehavior` is intentional to handle only one out of many signals after the gateway. I have changed the corresponding code in `IntermediateCatchEventActivityBehavior` to distinguish for the difference in handling. I also moved common signal event registration and checking logic to FlowNodeActivityBehavior super class as well.